### PR TITLE
SWATCH-2361: rhsm-conduit should forward the conversions.activity to HBI

### PR DESCRIPTION
**Pull request by Sgitario**
Jira issue: [SWATCH-2361](https://issues.redhat.com/browse/SWATCH-2361)

## Description
- forward the conversions.activity fact

## Testing

1.- podman-compose up
2.- start conduit service:

```
RHSM_USE_STUB=true DEV_MODE=true ./gradlew :swatch-system-conduit:bootRun
```

3.- sync the org ID:

```
http :8000/api/rhsm-subscriptions/v1/internal/rpc/syncOrg   x-rh-swatch-psk:placeholder   org_id=org123
```

4.- check the new message that is sent into the Kafka topic "platform.inventory.host-ingress":

```
{
        "org_id": "org123",
        "subscription_manager_id": "f665078c-bc8c-473d-b3c6-b4e315f2bd9e",
        "bios_uuid": "89462a07-32f8-4d75-a70c-77c2bd85a496",
        "ip_addresses": [
          "192.167.11.2",
          "192.168.1.1",
          "192.168.122.1",
          "10.0.0.1"
        ],
        "fqdn": "host1.test.com",
        "facts": [
          {
            "namespace": "rhsm",
            "facts": {
              "SYNC_TIMESTAMP": "2024-05-21T11:39:08.053973638Z",
              "IS_VIRTUAL": true,
              "SYSPURPOSE_SLA": "Premium",
              "MEMORY": 32,
              "conversions.activity": "conversion",
              "ARCHITECTURE": "x86_64",
              "VM_HOST": "hypervisor1.test.com",
              "SYSPURPOSE_UNITS": "Sockets",
              "RH_PROD": [
                "72"
              ],
              "orgId": "org123",
              "BILLING_MODEL": "standard"
            }
          }
        ],
        "system_profile": {
          "operating_system": {
            "major": 6,
            "minor": 3,
            "name": "RHEL"
          },
          "os_release": "6.3",
          "releasever": "8.0",
          "arch": "x86_64",
          "cores_per_socket": 2,
          "infrastructure_type": "virtual",
          "system_memory_bytes": 33543999488,
          "number_of_sockets": 2,
          "number_of_cpus": 3,
          "threads_per_core": 5,
          "owner_id": "f665078c-bc8c-473d-b3c6-b4e315f2bd9e",
          "is_marketplace": true,
          "network_interfaces": [
            {
              "ipv4_addresses": [
                "127.0.0.1"
              ],
              "mac_address": "00:00:00:00:00:00",
              "name": "lo"
            }
          ],
          "conversions": {
            "activity": true
          }
        },
        "stale_timestamp": "2024-05-23T11:39:08.053973638Z",
        "reporter": "rhsm-conduit",
        "provider_id": "123456",
        "provider_type": "aws"
      }
```

Note the new "conversions_activity" system profile fact.

### CODE REVIEW

File: swatch-system-conduit/schemas/inventory/hbi-system-profile.yaml [hunk 1]
```diff
@@ -37,4 +37,5 @@ properties:
       $ref: "hbi-network-interface.yaml"
   virtual_host_uuid:
     type: string
-
+  conversions_activity:
+    type: string
```
Comment (barnabycourt) [inline swatch-system-conduit/schemas/inventory/hbi-system-profile.yaml:41]: Per: https://github.com/RedHatInsights/inventory-schemas/pull/129/files#diff-952f7dbba2e7a294392a7ac96f3f9a505fff472c194ecf63763c221e7ef84fb5R759, I believe the value in the HBI system profile is a boolean

**Review by barnabycourt — CHANGES_REQUESTED** [review]
Additional tests should be included to validate if values such as "analysis" or null are specified in the candlepin facts, then the value in the system profile will be false/unset.

**Comment by Aurobinda55** [comment]
@Sgitario I am not able to see the fact(conversions_activity) in the system profile fatcs,

```
In [10]: app.host_inventory.rest_client.hosts_api.api_host_get_host_system_profile_by_id(["19bee6cb-4f27-4c2e-a4b3-7ef32d66127d"])
2024-05-14 04:17:40.320 [    INFO] [iqe.base.rest_client] REST: METHOD=GET, request_id=None, params=[]
Out[10]: 
{'count': 1,
 'page': 1,
 'per_page': 50,
 'results': [{'id': '19bee6cb-4f27-4c2e-a4b3-7ef32d66127d',
              'system_profile': {'ansible': None,
                                 'arch': 'x86_64',
                                 'basearch': 'string',
                                 'bios_release_date': 'string',
                                 'bios_vendor': 'string',
                                 'bios_version': 'string',
                                 'bootc_status': None,
                                 'captured_date': None,
                                 'cloud_provider': 'test cloud provider',
                                 'conversions': None,
                                 'cores_per_socket': 1,
                                 'cpu_flags': [],
                                 'cpu_model': None,
                                 'disk_devices': [],
                                 'dnf_modules': None,
                                 'enabled_services': [],
                                 'gpg_pubkeys': None,
                                 'greenboot_fallback_detected': None,
                                 'greenboot_status': None,
                                 'host_type': None,
                                 'infrastructure_type': 'virtual',
                                 'infrastructure_vendor': 'string',
                                 'insights_client_version': 'string',
                                 'insights_egg_version': 'string',
                                 'installed_packages': [],
                                 'installed_packages_delta': None,
                                 'installed_products': [],
                                 'installed_services': [],
                                 'intersystems': None,
                                 'is_marketplace': None,
                                 'katello_agent_running': False,
                                 'kernel_modules': [],
                                 'last_boot_time': datetime.datetime(2024, 5, 13, 10, 14, 23, 979466, tzinfo=tzlocal()),
                                 'mssql': None,
                                 'network_interfaces': [],
                                 'number_of_cpus': 1,
                                 'number_of_sockets': 1,
                                 'operating_system': {'major': 8,
                                                      'minor': 7,
                                                      'name': 'RHEL'},
                                 'os_kernel_version': '3.10.0',
                                 'os_release': 'string',
                                 'owner_id': 'be73358c-74f5-4a59-90ba-6052b00735a1',
                                 'public_dns': None,
                                 'public_ipv4_addresses': None,
                                 'releasever': 'string',
                                 'rhc_client_id': None,
                                 'rhc_config_state': None,
                                 'rhsm': None,
                                 'rpm_ostree_deployments': None,
                                 'running_processes': [],
                                 'sap': None,
                                 'sap_instance_number': None,
                                 'sap_sids': None,
                                 'sap_system': None,
                                 'sap_version': None,
                                 'satellite_managed': True,
                                 'selinux_config_file': None,
                                 'selinux_current_mode': None,
                                 'subscription_auto_attach': 'string',
                                 'subscription_status': 'string',
                                 'system_memory_bytes': 77824,
                                 'system_purpose': None,
                                 'system_update_method': None,
                                 'systemd': None,
                                 'threads_per_core': 1,
                                 'tuned_profile': None,
                                 'virtual_host_uuid': None,
                                 'yum_repos': []}}],
 'total': 1}

```

**Comment by Sgitario** [comment]
> @Sgitario I am not able to see the fact(conversions_activity) in the system profile fatcs,
> 
> ```
> In [10]: app.host_inventory.rest_client.hosts_api.api_host_get_host_system_profile_by_id(["19bee6cb-4f27-4c2e-a4b3-7ef32d66127d"])
> 2024-05-14 04:17:40.320 [    INFO] [iqe.base.rest_client] REST: METHOD=GET, request_id=None, params=[]
> Out[10]: 
> {'count': 1,
>  'page': 1,
>  'per_page': 50,
>  'results': [{'id': '19bee6cb-4f27-4c2e-a4b3-7ef32d66127d',
>               'system_profile': {'ansible': None,
>                                  'arch': 'x86_64',
>                                  'basearch': 'string',
>                                  'bios_release_date': 'string',
>                                  'bios_vendor': 'string',
>                                  'bios_version': 'string',
>                                  'bootc_status': None,
>                                  'captured_date': None,
>                                  'cloud_provider': 'test cloud provider',
>                                  'conversions': None,
>                                  'cores_per_socket': 1,
>                                  'cpu_flags': [],
>                                  'cpu_model': None,
>                                  'disk_devices': [],
>                                  'dnf_modules': None,
>                                  'enabled_services': [],
>                                  'gpg_pubkeys': None,
>                                  'greenboot_fallback_detected': None,
>                                  'greenboot_status': None,
>                                  'host_type': None,
>                                  'infrastructure_type': 'virtual',
>                                  'infrastructure_vendor': 'string',
>                                  'insights_client_version': 'string',
>                                  'insights_egg_version': 'string',
>                                  'installed_packages': [],
>                                  'installed_packages_delta': None,
>                                  'installed_products': [],
>                                  'installed_services': [],
>                                  'intersystems': None,
>                                  'is_marketplace': None,
>                                  'katello_agent_running': False,
>                                  'kernel_modules': [],
>                                  'last_boot_time': datetime.datetime(2024, 5, 13, 10, 14, 23, 979466, tzinfo=tzlocal()),
>                                  'mssql': None,
>                                  'network_interfaces': [],
>                                  'number_of_cpus': 1,
>                                  'number_of_sockets': 1,
>                                  'operating_system': {'major': 8,
>                                                       'minor': 7,
>                                                       'name': 'RHEL'},
>                                  'os_kernel_version': '3.10.0',
>                                  'os_release': 'string',
>                                  'owner_id': 'be73358c-74f5-4a59-90ba-6052b00735a1',
>                                  'public_dns': None,
>                                  'public_ipv4_addresses': None,
>                                  'releasever': 'string',
>                                  'rhc_client_id': None,
>                                  'rhc_config_state': None,
>                                  'rhsm': None,
>                                  'rpm_ostree_deployments': None,
>                                  'running_processes': [],
>                                  'sap': None,
>                                  'sap_instance_number': None,
>                                  'sap_sids': None,
>                                  'sap_system': None,
>                                  'sap_version': None,
>                                  'satellite_managed': True,
>                                  'selinux_config_file': None,
>                                  'selinux_current_mode': None,
>                                  'subscription_auto_attach': 'string',
>                                  'subscription_status': 'string',
>                                  'system_memory_bytes': 77824,
>                                  'system_purpose': None,
>                                  'system_update_method': None,
>                                  'systemd': None,
>                                  'threads_per_core': 1,
>                                  'tuned_profile': None,
>                                  'virtual_host_uuid': None,
>                                  'yum_repos': []}}],
>  'total': 1}
> ```

You need to create a Host in HBI with the system profile fact `conversions.activity: conversion` like I did in the stub service: https://github.com/RedHatInsights/rhsm-subscriptions/pull/3341/files#diff-d0dfdf9aac89aaddb01f04ff3427fa9348368c0d6f4a4b7a69590c8a3c331fbfR90

**Comment by Aurobinda55** [comment]
Verification steps :

- Created a physical host with fact- `{"conversions. activity": "conversion"}`
- Sync conduit 
- Checked system profile and got` 'conversions': {'activity': True},`
- Repeat step-1 and 2 and replace `conversion.activity: ''any"`
- Checked system profile and found `conversions': {'activity': False}`

Added automation https://gitlab.cee.redhat.com/insights-qe/iqe-rhsm-subscriptions-plugin/-/merge_requests/710

**Pull request merged**